BUILD

```diff
@@ -301,6 +301,7 @@ grpc_cc_library(
     language = "c++",
     public_hdrs = GPR_PUBLIC_HDRS,
     standalone = True,
+    visibility = ["//visibility:public"],
```

Comment:
how were the list of "public" libraries established? Did you just hand-picked those? How can we make sure the list is complete and there's nothing missing?

AFAIK not all the public libraries are actually in the /BUILD file, there are also some public items in some of the subdirectories.

Reply:
I marked all the standalone targets public by hand. Note this PR is just dealing with the top level BUILD file, and other targets in subdirectories are out of scope.

Reply:
I think we need to make sure that there's a clear mapping between `visibility = ["//visibility:public"]` and the `build` attribute in the script that generates build.yaml metadata from the bazel build:
https://github.com/grpc/grpc/blob/a43082112f5a2e4ef475c8b9c364ca08349e201b/tools/buildgen/extract_metadata_from_bazel_xml.py#L575
(there's clearly some relationship between the "public" targets and the ones marked as e.g. `build: all` and while doing this cleanup we should figure out what the source of truth is).